### PR TITLE
cmd/tailscale/cli: limit Darwin-only option to Darwin

### PR DESCRIPTION
**Pull request by noncombatant**
Hide the `-app-store` option with `HIDDEN: `.

Updates #6995

### CODE REVIEW

File: cmd/tailscale/cli/update.go [hunk 1]
```diff
@@ -44,7 +44,10 @@ var updateCmd = &ffcli.Command{
 		fs := newFlagSet("update")
 		fs.BoolVar(&updateArgs.yes, "yes", false, "update without interactive prompts")
 		fs.BoolVar(&updateArgs.dryRun, "dry-run", false, "print what update would do without doing it, or prompts")
-		fs.BoolVar(&updateArgs.appStore, "app-store", false, "check the App Store for updates, even if this is not an App Store install (for testing only!)")
+		if runtime.GOOS == "darwin" {
+			// TODO(cpalmer): Maybe Windows someday as well.
+			fs.BoolVar(&updateArgs.appStore, "app-store", false, "check the App Store for updates, even if this is not an App Store install (for testing only!)")
```
Comment (bradfitz) [inline cmd/tailscale/cli/update.go:49]: drop the exclamation mark. they're always more fun to write than they are to read.

**Review by bradfitz — APPROVED** [review]

File: cmd/tailscale/cli/update.go (hunk 1 above)
Comment (bradfitz) [reply cmd/tailscale/cli/update.go:49]: But what does "for testing only" mean? Testing by whom? Us? Should this be behind an envknob?

Or! Better! (if you want this to be available only if you know about it)

Remove the darwin conditional and instead make the description start with the magic phrase `"HIDDEN: "` and then it'll be omitted by our flag help printing code but the flag will still exist, if you know about it.

**Review by awly — APPROVED** [review]
TIL about `HIDDEN: ` flag prefix!

File: cmd/tailscale/cli/update.go (hunk 1 above)
Comment (noncombatant) [reply cmd/tailscale/cli/update.go:49]: Yes, testing by us. `HIDDEN: ` sounds like the way to go; new version coming up.

**Pull request merged**